index.js

```diff
@@ -0,0 +1,244 @@
+      );
+  });
+
+  // POST
+  server.post("/api/projects", (req, res) => {
+    const project = req.body;
+    if (!project.name || !project.comments) {
+      res
+        .status(400)
+        .json({ error: "This project needs a name and comments!" });
+    } else
+      db.insert(project)
+        .into("projects")
+        .then(ids => {
+          res.status(201).json(ids);
+        })
+        .catch(err =>
+          res
+            .status(500)
+            .json({ error: "Uh oh! The project could not be saved." })
+        );
+  });
+
+  // GET ID
+  server.get('/api/projects/:id', (req, res) => {
+    const id = req.params.id;
+     db("projects")
+        .select()
+        .where("id", id)
+        .then(projects => {
+            if(projects){
+                res.status(200).json(projects);
+            }else{
+                res.status(404).json({message: "Uh oh! There is no project with this ID!"});
+            }
+  
+        })
+        .catch(err => res.status(500).json(err));
+  });
+ 
+  // DELETE
+  server.delete("/api/projects/:id", (req, res) => {
+    const { id } = req.params;
+    db("projects")
+      .where({ id: id })
+      .del()
+      .then(count => {
+        if (count) {
+          res
+            .status(200)
+            .json({ message: "The project was successfully deleted!" });
+        } else {
+          res
+            .status(404)
+            .json({ message: "Uh oh! There is no project with this ID!" });
+        }
+      })
+      .catch(err =>
+        res.status(500).json({ error: "Uh oh! The project couldn't be deleted!" })
+      );
+});
+
+// PUT
+server.put("/api/projects/:id", (req, res) => {
+    const { id } = req.params;
+    const project = req.body;
+    if (!project.name || !project.comments) {
+      res
+        .status(400)
+        .json({ error: "This project needs a name and comments!" });
+    } else
+      db("projects")
+        .where({ id: id })
+        .update(project)
+        .then(count => {
+          if (count) {
+            res
+              .status(200)
+              .json({ message: "The project was successfully updated." });
+          } else {
+            res
+              .status(404)
+              .json({ message: "Uh oh! There is no project with this ID!" });
+          }
+        })
+        .catch(err =>
+          res
+            .status(500)
+            .json({ error: "Uh oh! The project couldn't be updated" })
+        );
+  });
+
+//========== ACTION ENDPOINTS ==========//
+// GET ACTIONS
+server.get("/api/actions", (req, res) => {
+    db("actions")
+      .then(actions => {
+        res.status(200).json(actions);
+      })
+      .catch(err =>
+        res
+          .status(500)
+          .json({ error: "Uh oh! The actions could not be retrieved." })
+      );
+  });
+
+// POST
+server.post("/api/actions", verifyProject, (req, res) => {
+    const action = req.body;
+    if (!action.comments || !action.notes) {
```

Comment:
There is no 'comments' column on your actions table either. This is confusing for someone who did not write the code.